ballerina/client.bal

```diff
@@ -357,760 +358,425 @@ public isolated client class Client {
-                                                    returns @display {label: "Organization Limits"}
-                                                    map<Limit>|Error {
-        string path = utils:prepareUrl([API_BASE_PATH, LIMITS]);
-        json res = check self.get(path);
-        return toMapOfLimits(res);
-    }
-
-    # Accesses records based on the specified object ID, can be used with external objects.
-    #
-    # + path - Resource path
-    # + return - JSON result if successful else or else `sfdc:Error`
-    #
-    # # Deprecated
-    # This function is deprecated as it returns a json response.
-    # Use the new and improved `getById(string sObject, string id)` function instead.
-    @display {label: "Get Record"}
-    @deprecated
-    isolated remote function getRecord(@display {label: "Resource Path"} string path)
-                                        returns @display {label: "Result"} json|Error {
-        json|http:ClientError response = self.salesforceClient->get(path);
-        if response is json {
-            return response;
+    // Added APIs
+    
+    # Retrieves the list of individual records that have been deleted within the given timespan.
+    #
+    # + sObjectName - SObject reference
+    # + startDate - Start date of the timespan.
+    # + endDate - End date of the timespan.
+    # + return - `DeletedRecordsResult` record if successful or else `error`
+    isolated remote function getDeleted(string sObjectName, time:Civil startDate, time:Civil endDate) 
+        returns DeletedRecordsResult|error {
+        string path = utils:prepareUrl([API_BASE_PATH, SOBJECTS, sObjectName, DELETED]);
+        string finalUrl = utils:addQueryParameters(path,{'start: check time:civilToString(removeDecimalPlaces(startDate)), 
+            end: check time:civilToString(removeDecimalPlaces(endDate))});
+        return check self.salesforceClient->get(finalUrl);
+    }
+
+    # Retrieves the list of individual records that have been deleted within the given timespan.
+    #
+    # + sObjectName - SObject reference
+    # + startDate - Start date of the timespan.
+    # + endDate - End date of the timespan.
+    # + return - `UpdatedRecordsResults` record if successful or else `error`
+    isolated remote function getUpdated(string sObjectName, time:Civil startDate, time:Civil endDate) 
+        returns UpdatedRecordsResults|error {
+        string path = utils:prepareUrl([API_BASE_PATH, SOBJECTS, sObjectName, UPDATED]);
+        string finalUrl = utils:addQueryParameters(path,{'start: check time:civilToString(removeDecimalPlaces(startDate)), 
+            end: check time:civilToString(removeDecimalPlaces(endDate))});
+        return check self.salesforceClient->get(finalUrl);
+    }
+
+    # Get the password information
+    #
+    # + userId - User ID
+    # + return - `boolean` if successful or else `error`
+    isolated remote function getPasswordInfo(string userId) returns boolean|error {
```

Comment:
If this is to check only password expiry, shall we name as `isPasswordExpired`? 
Do we really need to return an error?

Reply:
in cases of casting and network related errors, we need to return one right?